base/regex.jl

```diff
@@ -15,6 +15,8 @@ type Regex
     extra::Ptr{Void}
     ovec::Vector{Csize_t}
     match_data::Ptr{Void}
+    capture_name_to_idx::Dict{Symbol, Int}
+    idx_to_capture_name::Dict{Int, Symbol}
```

Comment:
i think it would be better to do the computation on-demand rather than caching it. a linear scan over the nametable will probably be the same time-performance as this dict lookup, but considerably less memory.